### ISSUE TITLE
[media-feature-colon-space-(before|after)] fails on @custom-selector

### ISSUE THREAD

**Issue by onigoetz** [stylelint/stylelint#359]
It seems the media-feature rule doesn't just check on  `@media` but also checks `@custom-selector` rules

But the rules that apply on one are not pretty on the other one

```
rules: {
      "media-feature-colon-space-after": [ 2, "always" ],
      "media-feature-colon-space-before": [ 2, "never" ],
}	
```

```css
@custom-selector :--enter :hover, :focus;
```

```
input.css
1:1	⚠  Expected single space after ":" (media-feature-colon-space-after) [stylelint]
1:1	⚠  Expected single space after ":" (media-feature-colon-space-after) [stylelint]
1:1	⚠  Expected single space after ":" (media-feature-colon-space-after) [stylelint]
1:1	⚠  Unexpected whitespace before ":" (media-feature-colon-space-before) [stylelint]
1:1	⚠  Unexpected whitespace before ":" (media-feature-colon-space-before) [stylelint]
```

**Comment by jeddy3** [comment]
> It seems the media-feature rule doesn't just check on @media but also checks @custom-selector rules

I think this rule should only be checking the whitespace around colon in media features. I think we should tighten it up so it excludes anything but @ media rules.

**Comment by jeddy3** [comment]
It looks like we don't need a new rule for space around the colon in custom selectors as spaces aren't allowed there:

" ... there must be no whitespace between : and <extension-name> or between $ and <ident-token> in the above definitions."

https://drafts.csswg.org/css-extensions/#custom-selectors

**Comment by davidtheclark** [comment]
Thanks for taking care of this one @jeddy3.
